### PR TITLE
Lesson10

### PR DESCRIPTION
**Pull request by NPaulGuy**

### CODE REVIEW

File: Lesson10/Task12.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Сделайте поле ввода, в которое пользователь вводит год (4 цифры), а скрипт определяет високосный ли год.
+ */
+if (
+    !empty($_POST['year']) 
+    && $_POST['year'] >= 1000
+    && $_POST['year'] < 10000
+) {
+    $year = trim($_POST['year']); 
+    if ($year % 4 === 0) {
```
Comment (ckomarov) [inline Lesson10/Task12.php:11]: Why 4?

File: Lesson10/Task12.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Сделайте поле ввода, в которое пользователь вводит год (4 цифры), а скрипт определяет високосный ли год.
+ */
+if (
```
Comment (ckomarov) [inline Lesson10/Task12.php:5]: What if condition is false?

File: Lesson10/Task12.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if ($year % 4 === 0) {
+        echo "Год високосный!";
+    } else {
+        echo "Год не високосный.";
+    }
+}
+?>
+<form method="post" name="inputForm">
+    <label>Год:</label><br>
+    <input type="text" name="year">
```
Comment (ckomarov) [inline Lesson10/Task12.php:20]: Maybe text type is not a good choice for year?

File: Lesson10/Task13.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    && preg_match('#^\d{2}\.\d{2}\.\d{4}$#', $_POST['date'])
+) {
+    $date = $_POST['date']; 
+    $arr = explode('.', $date);
+    echo $week[date("w", mktime(0, 0, 0, $arr[1], $arr[0], $arr[2]))];
+}
+?>
+<form method="post" name="inputForm">
+    <label>Дата:</label><br>
+    <input type="text" name="date">
```
Comment (ckomarov) [inline Lesson10/Task13.php:18]: The same as above

File: Lesson10/Task14.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    && preg_match('#^\d{4}-\d{2}-\d{2}$#', $_POST['date'])
+) {
+    $date = $_POST['date']; 
+    $arr = explode('-', $date);
+    echo $month[date("n", mktime(0, 0, 0, $arr[1], $arr[2], $arr[0]))];
+}
+?>
+<form method="post" name="inputForm">
+    <label>Дата:</label><br>
+    <input type="text" name="date">
```
Comment (ckomarov) [inline Lesson10/Task14.php:22]: Same

File: Lesson10/Task15.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * 	Сделайте форму, которая спрашивает две даты в формате '2025-12-31'. Первую дату запишите в переменную $date1, а вторую в $date2. Сравните, какая из введенных дат больше. Выведите ее на экран.
+ */
+if (
+    !empty($_POST['date1']) 
+    && !empty($_POST['date2'])
+    && preg_match('#^\d{4}-\d{2}-\d{2}$#', $_POST['date1'])
+    && preg_match('#^\d{4}-\d{2}-\d{2}$#', $_POST['date2'])
```
Comment (ckomarov) [inline Lesson10/Task15.php:9]: Can we check it without regular expression?

File: Lesson10/Task15.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    $date1 = $_POST['date1']; 
+    $date2 = $_POST['date2']; 
+    $arr1 = explode('-', $date1);
+    $arr2 = explode('-', $date2);
+    $date1 = mktime(0, 0, 0, $arr1[1], $arr1[2], $arr1[0]);
+    $date2 = mktime(0, 0, 0, $arr2[1], $arr2[2], $arr2[0]);
+    if ($date1 > $date2) {
+        echo "Больше Дата1: " . $_POST['date1'];
+    } elseif ($date1 < $date2) {
+        echo "Большое Дата2: " . $_POST['date2'];
```
Comment (ckomarov) [inline Lesson10/Task15.php:20]: Typo

File: Lesson10/Task20.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        if (date("w", mktime(0, 0, 0, $i, 13, $year)) == 5) {
+            echo date("d.m", mktime(0, 0, 0, $i, 13, $year)) . "<br>";
+        }
+    }
+}
+?>
+<form method="post" name="inputForm">
+    <label>Год:</label><br>
+    <input type="text" name="year">
+    <input type="submit" name="submitButton" value="Ввод">
```
Comment (ckomarov) [inline Lesson10/Task20.php:21]: Same as above

**Review by ckomarov — CHANGES_REQUESTED** [review]

File: Lesson10/Task12.php (hunk 1 above)
Comment (NPaulGuy) [reply Lesson10/Task12.php:11]: Lesson-2: Task 22: 
/**
 * В переменной $year хранится год. Определите, является ли он високосным (в таком году есть 29 февраля). 
 * **Год будет високосным в двух случаях: либо он делится на 4, но при этом не делится на 100, либо делится на 400.** (Year can be divided by 4 but not by 100 or can be divided by 400 (400 = 4 * 100). That's why we divide year only by 4 to know if it is leap-year)
 */

File: Lesson10/Task12.php (hunk 2 above)
Comment (NPaulGuy) [reply Lesson10/Task12.php:5]: If condition is false, it just prints the form.

File: Lesson10/Task14.php (hunk 5 above)
Comment (NPaulGuy) [reply Lesson10/Task14.php:22]: The condition of the task asks to input date in format YYYY-mm-dd, but types "date" and "datetime-local" have locked input format dd.mm.YYYY, so it is necessary to use different type, like "text".

File: Lesson10/Task15.php (hunk 6 above)
Comment (NPaulGuy) [reply Lesson10/Task15.php:9]: Regular expression helps to follow the conditions of the task: "input:date" and "input:datetime-local" don't fit it. So we use regular expression to check date format.

File: Lesson10/Task20.php (hunk 8 above)
Comment (NPaulGuy) [reply Lesson10/Task20.php:18]: If it is about typo, I couldn't see that.

**Review by ckomarov — APPROVED** [review]

**Pull request merged**